x-pack/filebeat/input/azureeventhub/connection_string.go

```diff
@@ -0,0 +1,132 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+
+package azureeventhub
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// ConnectionStringProperties are the properties of a connection string
+// as returned by [ParseConnectionString].
+type ConnectionStringProperties struct {
+	// Endpoint is the Endpoint value in the connection string.
+	// Ex: sb://example.servicebus.windows.net
+	Endpoint string
+
+	// EntityPath is EntityPath value in the connection string.
+	EntityPath *string
+
+	// FullyQualifiedNamespace is the Endpoint value without the protocol scheme.
+	// Ex: example.servicebus.windows.net
+	FullyQualifiedNamespace string
+
+	// SharedAccessKey is the SharedAccessKey value in the connection string.
+	SharedAccessKey *string
+
+	// SharedAccessKeyName is the SharedAccessKeyName value in the connection string.
+	SharedAccessKeyName *string
+
+	// SharedAccessSignature is the SharedAccessSignature value in the connection string.
+	SharedAccessSignature *string
+
+	// Emulator indicates that the connection string is for an emulator:
+	// ex: Endpoint=localhost:6765;SharedAccessKeyName=<< REDACTED >>;SharedAccessKey=<< REDACTED >>;UseDevelopmentEmulator=true
+	Emulator bool
+}
+
+// ParseConnectionString takes a connection string from the Azure portal and returns the
+// parsed representation.
+//
+// There are two supported formats:
+//
+//  1. Connection strings generated from the portal (or elsewhere) that contain an embedded key and keyname.
+//
+//  2. A connection string with an embedded SharedAccessSignature:
+//     Endpoint=sb://<sb>.servicebus.windows.net;SharedAccessSignature=SharedAccessSignature sr=<sb>.servicebus.windows.net&sig=<base64-sig>&se=<expiry>&skn=<keyname>"
+func parseConnectionString(connStr string) (ConnectionStringProperties, error) {
```

Comment:
nit (not needed now)  - Testing

I see this is moved from another place (+1 for isolation of logic). I see tests covering most of the parsing scenarios. But we could add dedicated tests focusing on `parseConnectionString`. Anyway this is not a blocker and could be a followup